ext/node/ops/handle_wrap.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use deno_core::error::ResourceError;
+use deno_core::op2;
+use deno_core::v8;
+use deno_core::GarbageCollected;
+use deno_core::OpState;
+use deno_core::ResourceId;
+
+pub struct AsyncId(i64);
+
+impl Default for AsyncId {
+  // `kAsyncIdCounter` should start at `1` because that'll be the id the execution
+  // context during bootstrap.
+  fn default() -> Self {
+    Self(1)
+  }
+}
+
+impl AsyncId {
+  // Increment the internal id counter and return the value.
+  fn next(&mut self) -> i64 {
+    self.0 += 1;
+    self.0
+  }
+}
+
+fn next_async_id(state: &mut OpState) -> i64 {
+  let async_id = state.borrow_mut::<AsyncId>().next();
+  async_id
+}
+
+#[op2(fast)]
+pub fn op_node_new_async_id(state: &mut OpState) -> f64 {
+  next_async_id(state) as f64
+}
+
+pub struct AsyncWrap {
+  provider: i32,
+  async_id: i64,
+}
+
+impl GarbageCollected for AsyncWrap {}
+
+impl AsyncWrap {
+  pub(crate) fn create(state: &mut OpState, provider: i32) -> Self {
+    let async_id = next_async_id(state);
+
+    Self { provider, async_id }
+  }
+}
+
+#[op2(base)]
+impl AsyncWrap {
+  #[getter]
+  fn provider(&self) -> i32 {
+    self.provider
+  }
+
+  #[fast]
+  fn get_async_id(&self) -> f64 {
+    self.async_id as f64
+  }
+
+  #[fast]
+  fn get_provider_type(&self) -> i32 {
+    self.provider
+  }
+}
+
+pub struct HandleWrap {
+  handle: Option<ResourceId>,
+}
+
+impl GarbageCollected for HandleWrap {}
+
+impl HandleWrap {
+  pub(crate) fn create(handle: Option<ResourceId>) -> Self {
+    Self { handle }
+  }
+}
+
+static ON_CLOSE_STR: deno_core::FastStaticString =
+  deno_core::ascii_str!("_onClose");
+
+#[op2(inherit = AsyncWrap)]
+impl HandleWrap {
+  #[constructor]
+  #[cppgc]
+  fn new(
+    state: &mut OpState,
+    #[smi] provider: i32,
+    #[smi] handle: Option<ResourceId>,
+  ) -> (AsyncWrap, HandleWrap) {
+    (
+      AsyncWrap::create(state, provider),
+      HandleWrap::create(handle),
+    )
+  }
+
+  #[reentrant]
+  fn close(
```

Comment:
Because this is meant to exactly implement the Node HandleWrap, I think it would be useful to include links into the corresponding C++ code.
Add here
```
// Ported from Node.js
// https://github.com/nodejs/node/blob/038d82980ab26cd79abe4409adc2fecad94d7c93/src/handle_wrap.cc#L65-L85
```